hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSAdmin.java

```diff
@@ -1648,40 +1647,46 @@ public int metaSave(String[] argv, int idx) throws IOException {
    * @throws IOException If an error while getting datanode report
    */
   public int printTopology() throws IOException {
-      DistributedFileSystem dfs = getDFS();
-      final DatanodeInfo[] report = dfs.getDataNodeStats();
-
-      // Build a map of rack -> nodes from the datanode report
-      HashMap<String, TreeSet<String> > tree = new HashMap<String, TreeSet<String>>();
-      for(DatanodeInfo dni : report) {
-        String location = dni.getNetworkLocation();
-        String name = dni.getName();
-        
-        if(!tree.containsKey(location)) {
-          tree.put(location, new TreeSet<String>());
-        }
+    DistributedFileSystem dfs = getDFS();
+    final DatanodeInfo[] report = dfs.getDataNodeStats();
+
+    // Build a map of rack -> nodes from the datanode report
+    HashMap<String, HashMap<String, String>> tree = new HashMap<String,
+        HashMap<String, String>>();
```

Comment:
nit: could you please replace this with?
```
Map<String, HashMap<String, String>> map = new HashMap<>();
```

Reply:
This was my mistake, I will update later.

Reply:
Sorry, this is my mistake, I will update later.